libraries/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -791,74 +940,28 @@ void _IotMqtt_ProcessSend( IotTaskPool_t pTaskPool,
         {
             pOperation->u.operation.status = IOT_MQTT_NETWORK_ERROR;
         }
-        else
+        /* DISCONNECT operations are considered successful upon successful transmission. */
+        else if( pOperation->u.operation.type == IOT_MQTT_DISCONNECT )
         {
-            /* DISCONNECT operations are considered successful upon successful
-             * transmission. In addition, non-waitable operations with no callback
-             * may also be considered successful. */
-            if( pOperation->u.operation.type == IOT_MQTT_DISCONNECT )
-            {
-                /* DISCONNECT operations are always waitable. */
-                IotMqtt_Assert( waitable == true );
+            /* DISCONNECT operations are always waitable. */
+            IotMqtt_Assert( waitable == true );
 
-                pOperation->u.operation.status = IOT_MQTT_SUCCESS;
-            }
-            else if( waitable == false )
+            pOperation->u.operation.status = IOT_MQTT_SUCCESS;
+        }
+        /* Non-waitable operations with no callback are also considered successful. */
+        else if( waitable == false )
+        {
+            if( pOperation->u.operation.notify.callback.function == NULL )
             {
-                if( pOperation->u.operation.notify.callback.function == NULL )
-                {
-                    pOperation->u.operation.status = IOT_MQTT_SUCCESS;
-                }
+                pOperation->u.operation.status = IOT_MQTT_SUCCESS;
             }
         }
```

Comment:
Note that [MISRA 2012 15.7 states](https://rules.sonarsource.com/c/RSPEC-126) _"All if...__else if__ constructs shall be terminated with an else statement"_.  It looks like those may have been pulled out in a previous revision as I don't see them there either.  No need to change anything for this review but this will need to be revisited for MISRA compliance later.